operator/pkg/manifest/shared.go

```diff
@@ -517,6 +524,85 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 	return string(out), nil
 }
 
+var defaultSetFlagConvertPaths = []string{
+	"meshConfig.defaultConfig.proxyMetadata",
+}
+
+// convertDefaultIOPMapValues converts default map[string]string values into string.
+func convertDefaultIOPMapValues(outYAML string, setFlags []string) (string, error) {
+	return convertIOPMapValues(outYAML, setFlags, defaultSetFlagConvertPaths)
+}
+
+// convertIOPMapValues converts certain paths of map[string]string values into string.
+func convertIOPMapValues(outYAML string, setFlags []string, convertPaths []string) (string, error) {
+	for _, setFlagConvertPath := range convertPaths {
+		if containParentPath(setFlags, setFlagConvertPath) {
+			var (
+				converter              = map[string]interface{}{}
+				convertedProxyMetadata = map[string]string{}
+				subPaths               = strings.Split(setFlagConvertPath, ".")
+			)
+
+			if err := yaml.Unmarshal([]byte(outYAML), &converter); err != nil {
+				return outYAML, err
+			}
+			originMap, ok := converter["spec"].(map[string]any)
+			if !ok {
+				return outYAML, nil
+			}
+
+			for index, subPath := range subPaths {
+				if _, ok := originMap[subPath].(map[string]any); !ok {
+					return outYAML, fmt.Errorf("can not convert subPath %s in setFlag path %s",
+						subPath, setFlagConvertPath)
+				}
+
+				if index == len(subPaths)-1 {
+					for key, value := range originMap[subPath].(map[string]any) {
+						switch t := value.(type) {
+						case int64:
+							convertedProxyMetadata[key] = strconv.FormatInt(value.(int64), 10)
+						case bool:
+							convertedProxyMetadata[key] = strconv.FormatBool(value.(bool))
+						case float64:
+							convertedProxyMetadata[key] = strconv.FormatInt(int64(value.(float64)), 10)
+						case string:
+							convertedProxyMetadata[key] = value.(string)
+						default:
+							return outYAML, fmt.Errorf("not supported type %s", t)
+						}
+					}
+					originMap[subPath] = convertedProxyMetadata
+				} else {
+					originMap = originMap[subPath].(map[string]any)
+				}
+			}
+
+			convertedYaml, err := yaml.Marshal(converter)
+			if err != nil {
+				return outYAML, err
+			}
+			return string(convertedYaml), nil
+		}
+	}
+
+	return outYAML, nil
+}
+
+// containParentPath checks if setFlags contain parent path.
+func containParentPath(setFlags []string, parentPath string) bool {
+	ret := false
+	for _, sf := range setFlags {
+		p, _ := getPV(sf)
+		if strings.Contains(p, parentPath) {
+			ret = true
+			break
+		}
+		// if set multiple times, return last set value
```

Comment:
so according to current implementation, it seems that there still be bug if multiple `meshConfig.defaultConfig.proxyMetadata` environment variables setting with boolean, am I right?

Reply:
No, this comment is a bit misleading, I am going to remove this comment, currently all bool values in proxyMeta will be unmarshalled correctly.